docs/gatsby-node.js

```diff
@@ -1,73 +1,159 @@
+    });
+
+    createNodeField({
+      node,
+      name: "description",
+      value: node.frontmatter.description || node.description || metaFileData.description,
+    });
+
+    // creates a "collection" field to make it easier to filter nodes created by
+    // gatsby-source-filesystem instead of awkwardly filtering by file path
+    // https://github.com/gatsbyjs/gatsby/issues/1634#issuecomment-388899348
+
+    createNodeField({
+      node,
+      name: "collection",
+      value: parent.sourceInstanceName,
+    });
+
+    // Create a similar field that should be identical
+    // for all pages that are tabs in one folder
+    createNodeField({
+      node,
+      name: "tabCollection",
+      value: hasTabs ? parent.relativeDirectory : null,
+    });
+
+    // Make title the same for all pages in tabs
+    const title = hasTabs ? metaFileData.title : node.frontmatter.title;
+    createNodeField({
+      node,
+      name: "title",
+      value: hasTabs ? metaFileData.title : node.frontmatter.title,
+    });
 
-  const getLinkForComponents = () => {
-    if (dir.startsWith("/03-components")) {
-      if (dir.match(/primitive/)) {
-        return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/primitives/${title}`;
+    const getLinkForComponents = () => {
+      if (dir.startsWith("/03-components")) {
+        if (dir.match(/primitive/)) {
+          return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/primitives/${title}`;
+        }
+        if (dir.match(/grid/)) {
+          return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/utils/Grid`;
+        }
+        return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/${title}`;
       }
-      if (dir.match(/grid/)) {
-        return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/utils/Grid`;
+      return "";
+    };
+
+    createNodeField({
+      node,
+      name: "headerLink",
+      value: getLinkForComponents() || node.frontmatter.headerLink || metaFileData.headerLink,
+    });
+
+    if (node.fields.collection === "documentation") {
+      let documentTrail;
+
+      if (hasTabs && path.basename(node.fileAbsolutePath).startsWith("01-")) {
+        documentTrail = getDocumentTrail(node.fields.slug);
+      } else {
+        documentTrail = getDocumentTrail(getParentUrl(node.fields.slug));
+        documentTrail.push({
+          name: node.frontmatter.title,
+          url: node.fields.slug,
+        });
       }
-      return `https://github.com/kiwicom/orbit/tree/master/packages/orbit-components/src/${title}`;
-    }
-    return "";
-  };
 
-  createNodeField({
-    node,
-    name: "headerLink",
-    value: getLinkForComponents() || node.frontmatter.headerLink || metaFile.headerLink,
-  });
+      createNodeField({
+        node,
+        name: "trail",
+        value: documentTrail,
+      });
+    } else {
```

Comment:
i suppose you do not need that `else` block.
You have to create node field on every node, otherwise gatsby will fail, but you can omit the `if` in that case and just have something like that 🤔 
```
createNodeField({
        node,
        name: "trail",
        value: documentTrail || [],
      });
```

or even better to rewrite it in a function, without usage of let, more declarative.